encoding/json/inbound.go

```diff
@@ -41,9 +41,12 @@ type jsonHandler struct {
 }
 
 func (h jsonHandler) Handle(ctx context.Context, treq *transport.Request, rw transport.ResponseWriter) error {
+	treq.Encoding = Encoding
+	// TODO(abg): Should we fail requests if Rpc-Encoding does not match?
```

Comment:
Perhaps - this would let us avoid attempting to encode/decode instead of trying to do the work and failing. It adds a bit of overhead on the wire, but is probably ok. 

@kriskowal @prashantv @akshayjshah @blampe what do you guys think?